modules/ximgproc/include/opencv2/ximgproc/fourier_descriptors.hpp

```diff
@@ -0,0 +1,114 @@
+            @param dist distance between src and dst after matching.
+            @param fdContour false then src and dst are contours and true src and dst are fourier descriptors.
+            */
+            CV_WRAP void estimateTransformation(InputArray src, InputArray dst, OutputArray _alphaPhiST, double &dist , bool fdContour = false);
+            /** @brief set number of Fourier descriptors used in estimateTransformation
+
+            @param n number of Fourier descriptors equal to number of contour points after resampling.
+            */
+            CV_WRAP void setCtrSize(int n);
+            /** @brief set number of Fourier descriptors when estimateTransformation used vector<Point>
+
+            @param n number of fourier descriptors used for optimal curve matching.
+            */
+            CV_WRAP void setFDSize(int n);
+            /**
+            @returns number of fourier descriptors
+            */
+            CV_WRAP int getCtrSize() { return ctrSize; };
+            /**
+            @returns number of fourier descriptors used for optimal curve matching
+            */
+            CV_WRAP int getFDSize() { return fdSize; };
+        };
+        /**
+        * @brief   Fourier descriptors for planed closed curves
+        *
+        * For more details about this implementation, please see @cite PersoonFu1977
+        *
+        * @param   _src   contour type vector<Point> , vector<Point2f>  or vector<Point2d>
+        * @param   _dst   Mat of type CV_64FC2 and nbElt rows A VERIFIER
+        * @param   nbElt number of rows in _dst or getOptimalDFTSize rows if nbElt=-1
+        * @param   nbFD number of FD return in _dst _dst = [FD(1...nbFD/2) FD(nbFD/2-nbElt+1...:nbElt)]
+        *
+        */
+        CV_EXPORTS_W void fourierDescriptor(InputArray _src, OutputArray _dst, int nbElt=-1,int nbFD=-1);
+        /**
+        * @brief   transform a contour
+        *
+        * @param   _src   contour or Fourier Descriptors if fd is true
+        * @param   _t   transform Mat given by estimateTransformation
+        * @param   _dst   Mat of type CV_64FC2 and nbElt rows
+        * @param   fdContour true _src are Fourier Descriptors. fdContour false _src is a contour
+        *
+        */
+        CV_EXPORTS_W void transform(InputArray _src, InputArray _t,OutputArray _dst, bool fdContour=true);
+        /**
+        * @brief   Contour sampling .
+        *
+        * @param   _src   contour type vector<Point> , vector<Point2f>  or vector<Point2d>
+        * @param   _out   Mat of type CV_64FC2 and nbElt rows
+        * @param   nbElt number of points in _out contour
+        *
+        */
+        CV_EXPORTS_W void contourSampling(InputArray _src, OutputArray _out, int nbElt);
+
+        CV_EXPORTS_W Ptr<ContourFitting> createContourFitting(int ctr = 1024, int fd = 16);
```

Comment:
Use `create` here instead of `createContourFitting`. `ContourFitting::createContourFitting()` looks redundant in user code.